docs/streams/upgrade-guide.html

```diff
@@ -72,6 +72,42 @@ <h1>Upgrade Guide and API Changes</h1>
         More details about the new config <code>StreamsConfig#TOPOLOGY_OPTIMIZATION</code> can be found in <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-295%3A+Add+Streams+Configuration+Allowing+for+Optional+Topology+Optimization">KIP-295</a>.
     </p>
 
+    <h3><a id="streams_api_changes_240" href="#streams_api_changes_240">Streams API changes in 2.4.0</a></h3>
+
+    <!-- Placeholder KIP-213 -->
+
+    <!-- Placeholder KIP-307 -->
+    <!-- Placeholder KIP-479 -->
+
+    <!-- Placeholder KIP-429 -->
+
+    <!-- Placeholder KIP-444 -->
+    <!-- Placeholder KIP-471 -->
+
+    <p>
+        Kafka Streams <code>test-utils</code> got improved via
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-470%3A+TopologyTestDriver+test+input+and+output+usability+improvements">KIP-470</a>
+        to simplify testing with the <code>TopologyTestDriver</code>.
+        We deprecated <code>ConsumerRecordFactory</code>, <code>TopologyTestDriver#pipeInput()</code>,
+        <code>OutputVerifier</code>, as well as <code>TopologyTestDriver#readOutput()</code> and replace them with
+        <code>TestInputTopic</code> and <code>TestOutputTopic</code>, respectively.
+        We also introduced a new class <code>TestRecord</code> that simplifies assertion code.
+        Additionally, a few minor API improvements were applied.
```

Comment:
I'd suggest either just drop this sentence and refer users to the ref link for more details, or just leave out all the details in text here. Since it adds few extra information than possible confusions.